core/src/main/java/org/apache/iceberg/rest/HTTPClient.java

```diff
@@ -195,33 +195,34 @@ private <T> T execute(
   }
 
   @Override
-  public void head(String path, Consumer<ErrorResponse> errorHandler) {
-    execute(Method.HEAD, path, null, null, errorHandler);
+  public void head(String path, Map<String, String> headers, Consumer<ErrorResponse> errorHandler) {
+    execute(Method.HEAD, path, null, null, headers, errorHandler);
   }
 
   @Override
-  public <T extends RESTResponse> T get(String path, Class<T> responseType,
+  public <T extends RESTResponse> T get(String path, Class<T> responseType, Map<String, String> headers,
```

Comment:
One thing you might want to consider is wrapping some of these parameters (i.e. headers and errorHandler) in some sort of wrapper that we can extend to further scope requests within sessions.  Just thinking something on the order of:

```java
RequestContext {
    Supplier<Map<String, String>> requestHeaders() 
    void handleResponseHeaders(Consumer<Map<String,String>>)
    Consumer<ErrorResponse> errorHandler()
    }
```

I feel like it will be hard to extend the current pattern if we need to propagate information back or retry requests (e.g. the 419 case) without being able to capture the current state.